baseplate/server/queue_consumer.py

```diff
@@ -116,7 +132,7 @@ class QueueConsumerFactory(abc.ABC):
     """
 
     @abc.abstractmethod
-    def build_pump_worker(self, work_queue: queue.Queue) -> PumpWorker:
+    def build_pump_worker(self, work_queue: WriteableQueueLike) -> PumpWorker:
```

Comment:
I don't think you actually need these changes, `build_pump_worker` is only called in the "server" and it always passes a `queue.Queue`.